Sources/ASTVisitor.swift

```diff
@@ -0,0 +1,284 @@
+public protocol ExpressionVisitor {
+    associatedtype ExpressionResult
+    // Literal
```

Comment:
**Memo:** We talked and decided not to introduce `LiteralVisitor` and `OperationVisitor` because then `Literal` and `Operation` must `accept` an `ExpressionVisitor`, neither `LiteralVisitor` nor `OperationVisitor`, because `Literal` and `Operation` are subtypes of `Expression`.